examples/app_drive_mounted_s3/app.py

```diff
@@ -0,0 +1,165 @@
+
+        print(f"Drive1(id={self.drive_1.id}) was mounted successfully!")
+
+        # -------- Now check the second drive in this work --------
+
+        drive_2_filenames = set()
+        for filename in glob.iglob(f"{self.drive_2_mountdir}**/**", recursive=True):
+            print(filename)
+            drive_2_filenames.add(filename)
+
+        # ensure that the file exists in the printed list
+        assert self.drive_2_checkfile in drive_2_filenames
+        # assert that content size is reported and not just inode metadata size
+        assert os.path.getsize(self.drive_2_checkfile) > 1_000  # bytes
+        # verify that we can open and read the contents without errors
+        with open(self.drive_2_checkfile, "rb") as f:
+            f.read()
+
+        print(f"Drive2(id={self.drive_2.id}) was mounted successfully!")
+
+        # -------- Done! --------
+        self.done = True
+
+
+class Work2(L.LightningWork):
+    def __init__(self, drive: "Drive", *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # you must assign the drive to an attribute on the work in the __init__ method
+        # in order for it to be mounted when running on the cloud runtime.
+        self.drive = drive
+
+        # signal to indicate if work has completed executing or not
+        self.done = None
+
+    def run(self, check_file: str):
+        print(f"verifying that {check_file} exists")
+        assert os.path.exists(check_file)
+
+        print(f"verifying that the byte size of {check_file} is reasonably reported & not just the inode metadata")
+        assert os.path.getsize(check_file) > 1_000  # bytes
+
+        print(f"verifying that we can open and read the contents of {check_file} without errors")
+        with open(check_file, "rb") as f:
+            f.read()
+
+        print("verifications complete! Exiting work run...")
+        self.done = True
+
+
+class Flow(L.LightningFlow):
+    def __init__(self):
+        super().__init__()
+
+        # Work1 contains Drive definitions in the __init__() method
+        self.work_1 = Work1(parallel=True)
+
+        # Work 2 will pass a drive we define in the flow into the work as an init argument
+        self.work2_drive_root_folder = "/tmp/drive1/datadir/"
+        # The directory must exist when initializing the flow and work, even if it is empty.
+        # We will create, and mount to this path automatically when `Work.run()` is called.
+        os.makedirs(self.work2_drive_root_folder, exist_ok=True)
+
+        # create the drive and initialize Work2 with the drive object
+        drive = Drive("s3://ryft-public-sample-data/twitter/", root_folder=self.work2_drive_root_folder)
+        self.work_2 = Work2(drive, parallel=True)
+
+        # we will call Work2.run() with a file path to check to ensure that it is available and mounted.
+        self.work2_drive_checkfile = os.path.join(self.work2_drive_root_folder, "twitter2015.tar.gz")
+
+        self.start_time = None
```

Comment:
Could you shortly explain what this app does? 

I  would expect to see an example/test that the Drive is mounted to work_1 and writes some files, then work_2 to read the data of work_1.

Reply:
This doesn't work with s3 drives. This is read only.